### PR TITLE
Navbar: add tab index to logo and trigger productLogoClick with keyboard

### PR DESCRIPTION
**Pull request by ausnewland11**
## Description

Added tab index to the e-builder log so it can now be tabbed to. Also added a keydown function so that when the enter key is hit on the logo, it will copy the functionality of clicking it. 

References [#2040](https://github.com/trimble-oss/modus-web-components/issues/2040) 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: stencil-workspace/src/components/modus-navbar/product-logo/modus-navbar-product-logo.tsx [hunk 1]
```diff
@@ -8,8 +8,17 @@ export const ModusNavbarProductLogo: FunctionalComponent<{
 }> = ({ logos, onClick }) => {
   const { primary, secondary } = logos || {};
 
+const logoKeydownHandler = (event: KeyboardEvent) => {
```
Comment (cjwinsor) [inline stencil-workspace/src/components/modus-navbar/product-logo/modus-navbar-product-logo.tsx:11]: Can you update as follows.  I think its overdue that we start handling space key press for accessibility. 

```jsx
  const handleKeyDown = (event) => {
    if (event.key !== 'Enter' || event.key !== ' ') {
      return;
    }
    
    if (onClick) {
      onClick(event);
    }
  };
```

File: stencil-workspace/src/components/modus-navbar/product-logo/modus-navbar-product-logo.tsx [hunk 2]
```diff
@@ -8,8 +8,17 @@ export const ModusNavbarProductLogo: FunctionalComponent<{
...
+    return;
+  }
+  if (onClick) {
+    onClick(event);
+  }
+};
+
   return (
-    <div onClick={onClick} class="product-logo">
+    <div onClick={onClick} onKeyDown={logoKeydownHandler} tabindex="0" class="product-logo">
```
Comment (cjwinsor) [inline stencil-workspace/src/components/modus-navbar/product-logo/modus-navbar-product-logo.tsx:21]: Can you add `role="button"` to indicate that the div is interactive.  Also can you add an `aria-label="Logo"` while you're here.  Should help give some context for accessibility.

**Review by cjwinsor — CHANGES_REQUESTED** [review]
Can you also update the storybook `modus-navbar.stories.tsx` so that `parameters.actions.handles` array contains `productLogoClick` so we can validate the event is fired from storybook.

**Comment by ausnewland11** [comment]
The error message for the 'lint, build, and test' doesn't look like an issue to me but let me know what you think @cjwinsor

**Review by cjwinsor — APPROVED** [review]

**Pull request merged**